lib/models/asyncapi.js

```diff
@@ -234,6 +234,103 @@ class AsyncAPIDocument extends Base {
   traverseSchemas(callback, schemaTypesToIterate) {
     traverseAsyncApiDocument(this, callback, schemaTypesToIterate);
   }
+
+  /**
+   * Converts a valid AsyncAPI document to a JavaScript Object Notation (JSON) string.
+   * A stringified AsyncAPI document using this function should be parsed via the AsyncAPIDocument.parse() function - the JSON.parse() function is not compatible.
+   * 
+   * @param {AsyncAPIDocument} doc A valid AsyncAPIDocument instance.
+   * @param {(number | string)=} space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
+   * @returns {string}
+   */
+  static stringify(doc, space) {
+    return JSON.stringify(doc.json(), refReplacer(), space);
+  }
+
+  /* eslint-disable sonarjs/cognitive-complexity */
+  /**
+   * Converts a valid stringified AsyncAPIDocument instance into an object.
+   * 
+   * @param {string} doc A valid stringified AsyncAPIDocument instance.
+   * @returns {AsyncAPIDocument}
+   */
+  static parse(doc) { // NOSONAR
+    const objToPath = new Map();
+    const pathToObj = new Map();
+    const parsedJSON = JSON.parse(doc);
+    
+    function traverse(parent, field) {
+      let objOrPath = parent;
+      let path = '$ref:$';
+  
+      if (field !== undefined) {
+        // here can be string with `$ref` prefix or normal value 
+        objOrPath = parent[String(field)];
+        const concatenatedPath = field ? `.${  field}` : '';
+        path = objToPath.get(parent) + (Array.isArray(parent) ? `[${field}]` : concatenatedPath);
+      }
+  
+      objToPath.set(objOrPath, path);
+      pathToObj.set(path, objOrPath);
+      
+      const ref = pathToObj.get(objOrPath);
+      if (ref) {
+        parent[String(field)] = ref;
+      }
+      if (objOrPath === '$ref:$' || ref === '$ref:$') {
+        parent[String(field)] = parsedJSON;
+      }
+  
+      // traverse all keys, only if object is array/object
+      if (objOrPath === Object(objOrPath)) {
+        for (const f in objOrPath) {
+          traverse(objOrPath, f);
+        }
+      }
+    }
```

Comment:
What is the reason behind this function? 🤔

Reply:
If we call the `stringify` function then later each circular reference is stored as a path to the object (we use JSON Pointer for this) - check example in the PR's description. This function just parses the whole string and looks for the given paths and turns them into reference :) This can be compared to the `json-schema-ref-parser` library, but much simpler and only looks for references in the given object and not outside :)

Reply:
Got it 👍